### PR TITLE
[gdal] Update to 3.4.1

### LINKED ISSUE #22381: [gdal] Build error using latest port 3.4.0#6
**Host Environment**
- OS: Windows 10 21H1
- Compiler: VS2019

**To Reproduce**
`./vcpkg install gdal[core,hdf5,netcdf,postgresql,mysql-libmariadb,libspatialite,supported-default-features]:x64-windows-static-md`

**Failure logs** 
Error: Building package gdal:x64-windows-static-md failed with: BUILD_FAILED

`link /nologo /dll /INCLUDE:OSRValidate  /INCLUDE:OGR_G_GetPointCount  /INCLUDE:OGRRegisterAll /INCLUDE:GDALSimpleImageWarp  /INCLUDE:GDALReprojectImage  /INCLUDE:GDALComputeMedianCutPCT  /INCLUDE:GDALDitherRGB2PCT  /INCLUDE:OCTNewCoordinateTransformation port\*.obj gcore\*.obj alg\*.obj frmts\o\*.obj ogr\ogrsf_frmts\ogrsf_frmts.lib ogr\ogr.lib gnm\*.obj gnm\gnm_frmts\o\*.obj third_party\o\*.obj apps\commonutils.obj apps\gdalinfo_lib.obj apps\gdal_translate_lib.obj apps\gdalwarp_lib.obj apps\ogr2ogr_lib.obj  apps\gdaldem_lib.obj apps\nearblack_lib.obj apps\gdal_grid_lib.obj apps\gdal_rasterize_lib.obj apps\gdalbuildvrt_lib.obj  apps\gdalmdiminfo_lib.obj apps\gdalmdimtranslate_lib.obj    D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libexpatdMD.lib  D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libpq.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libpgcommon.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libpgport.lib Secur32.lib Shell32.lib Advapi32.lib Crypt32.lib Gdi32.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libcrypto.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libssl.lib          D:\projects\_latest\vcpkg\installed\x64-windows-static-md\debug\lib\jpeg.lib /libpath:D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib netcdf.lib libhdf5_hl_D.lib libhdf5_D.lib libszip_D.lib zlibd.lib libcurl-d.lib winmm.lib ws2_32.lib zlibd.lib advapi32.lib crypt32.lib    D:\projects\_latest\vcpkg\installed\x64-windows-static-md\debug\lib\tiffd.lib /libpath:D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib proj_d.lib ole32.lib shell32.lib sqlite3.lib tiffd.lib zlibd.lib jpeg.lib lzmad.lib zstd_staticd.lib libcurl-d.lib winmm.lib ws2_32.lib zlibd.lib advapi32.lib crypt32.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/sqlite3.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/librttopo.lib  D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libmariadb.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/geos_cd.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/geosd.lib /libpath:D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib libhdf5_D.lib libszip_D.lib zlibd.lib      D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libcurl-d.lib wsock32.lib wldap32.lib winmm.lib    legacy_stdio_definitions.lib odbc32.lib odbccp32.lib user32.lib  D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libpng16d.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/zlibd.lib   D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/openjp2.lib            D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/webpd.lib   D:\projects\_latest\vcpkg\installed\x64-windows-static-md\debug\lib\freexl.lib    D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/libxml2.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/iconv.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/charset.lib D:/projects/_latest/vcpkg/installed/x64-windows-static-md/debug/lib/lzmad.lib       ws2_32.lib          kernel32.lib psapi.lib wbemuuid.lib gcore\Version.res  /out:gdal_d.dll /implib:gdal_i_d.lib   /debug`

`LINK : fatal error LNK1104: cannot open file 'D:\projects\_latest\vcpkg\installed\x64-windows-static-md\debug\lib\librttopo.lib'`

**Additional context**

Static build failed, due to a bug in `dependency_win.cmake` (wrong double quotes in line 235, 236).

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?  
  Updates port gdal to version 3.4.1.
  Revises the dependency import and export in particular for the `nmake` build:
   - Use pkg-config where available
     In particular, use it for spatialite (also for the autotools build). Fixes #22381.
   - Link openssl libs only with postgresql. (Curl would request them via pkg-config.) Fixes https://github.com/microsoft/vcpkg/pull/22370#issuecomment-1006624723.
   - Link libgeotiff also in static builds.
   - Don't enable hidden dependencies (missing in vcpkg.json) unless feature `system-libraries` is enabled (similar to the autotools build, for compatibility, but really not recommended).
   - Drop giflib dependency which is not used with nmake.
   - Add explicit libiconv dependency because it is always installed with libxml2.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

**Comment by sphoto** [comment]
The static build (x64-windows-static-md) failed when using libspatialite, due to a bug when setting SPATIALITE_LIBRARY_REL/DBG variables in `dependency_win.cmake`. I already opened an issue for that.

**Comment by dg0yt** [comment]
> The static build (x64-windows-static-md) failed when using libspatialite, due to a bug when setting SPATIALITE_LIBRARY_REL/DBG variables in `dependency_win.cmake`. I already opened an issue for that.

#22370. It is not a regression.

**Comment by LilyWangLL** [comment]
Fixes #22381

**Comment by sphoto** [comment]
The openssl libs are optional dependencies and not needed when building just `gdal[core]`, but since they are required in `dependency_win.cmake`, the build fails when they are not available.

Those packages will be installed on a fresh system. There is no openssl.

`vcpkg install gdal[core]:x64-windows`
`Computing installation plan...`
`The following packages will be built and installed:`
`  curl[core,non-http,schannel,ssl,sspi]:x64-windows -> 7.80.0`
`  expat[core]:x64-windows -> 2.4.1`
`  gdal[core]:x64-windows -> 3.4.0#6`
`  geos[core]:x64-windows -> 3.10.0`
`  giflib[core]:x64-windows -> 5.2.1#1`
`  libgeotiff[core]:x64-windows -> 1.7.0`
`  libiconv[core]:x64-windows -> 1.16#11`
`  libjpeg-turbo[core]:x64-windows -> 2.1.2`
`  liblzma[core]:x64-windows -> 5.2.5#4`
`  libpng[core]:x64-windows -> 1.6.37#16`
`  libwebp[core,nearlossless,simd,unicode]:x64-windows -> 1.2.1`
`  libxml2[core]:x64-windows -> 2.9.12#4`
`  nlohmann-json[core]:x64-windows -> 3.10.4`
`  openjpeg[core]:x64-windows -> 2.4.0`
`  proj4[core,net,tiff]:x64-windows -> 8.2.1`
`  sqlite3[core,tool]:x64-windows -> 3.37.0#1`
`  tiff[core,jpeg,lzma,zip]:x64-windows -> 4.3.0#3`
`  vcpkg-cmake[core]:x64-windows -> 2021-12-20`
`  vcpkg-cmake-config[core]:x64-windows -> 2021-12-28`
`  vcpkg-pkgconfig-get-modules[core]:x64-windows -> 2021-04-02#1`
`  zlib[core]:x64-windows -> 1.2.11#13`
`  zstd[core]:x64-windows -> 1.5.0#1`
`Additional packages (*) will be modified to complete this operation.`

`LINK : fatal error LNK1104: cannot open file 'D:\projects\_latest\vcpkg\installed\x64-windows\debug\lib\libcrypto.lib'`

**Comment by dg0yt** [comment]
Thanks for the hint @sphoto. I guess you just spotted another problem with the explicit assumptions about static dependencies. I'm still dealing with passing the libspatialite usage requirements along the static build chain. For windows builds, I need to wait for 

(And in paralllel, I'm experimenting with gdal's coming CMake build, but this also needs work.)

**Comment by dg0yt** [comment]
Now with major rewrite of dependency lookup for nmake, using pkg-config where possible:
- [x] CI run with all features except `system-libraries` (hidden dependencies): [x86-windows](https://dev.azure.com/vcpkg/public/_build/results?buildId=65566&view=logs&j=878666d5-db33-5b27-9e7d-b0c7ee352005&t=af38c879-95b0-57f5-5080-aaa522621fea)
- [ ] ...

**Comment by dg0yt** [comment]
Gentle ping for review.

**Review by LilyWangLL — APPROVED** [review]

File: ports/gdal/portfile.cmake [hunk 1]
```diff
@@ -13,115 +13,53 @@ endif()
...
-        PNG_EXTERNAL_LIB=1
-        "PNGDIR=${PNG_INCLUDE_DIR}"
-        "ZLIB_INC=-I${ZLIB_INCLUDE_DIR}"
-        ZLIB_EXTERNAL_LIB=1
-        MSVC_VER=1900
-        )
+    set(NMAKE_OPTIONS
+        "DATADIR=${CURRENT_PACKAGES_DIR}/share/gdal"
+        "HTMLDIR=${CURRENT_PACKAGES_DIR}/share/gdal/html"
+        "MSVC_VER=1900"
```
Comment (JackBoosY) [inline ports/gdal/portfile.cmake:35]: Should we force this version?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/gdal/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/gdal/portfile.cmake:35]: You need to ask this question to someone else. It was there before, I don't use MSVC. So it is not a point for this PR to deal with.

**Review by JackBoosY — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**